typings/index.d.ts

```diff
@@ -0,0 +1,53 @@
+declare module 'out-of-band-cache' {
+  type JSONSerializable = Object | string | number | boolean | Date;
+
+  type GetResult = (value: JSONSerializable) => boolean;
+
+  interface SharedCacheProps {
+    maxAge?: number;
+    maxStaleness?: number;
+    shouldCache?: (getResultFn: GetResult) => boolean;
+  }
+
+  type UpdateFn = (
+    key: string,
+    staleValue: JSONSerializable
+  ) => Promise<JSONSerializable>;
+
+  interface CacheType {
+    init: (...args: any) => Promise<any>;
+    get: (...args: any) => Promise<any>;
```

Comment:
Isn't there a more defined interface than this?

Reply:
How about:

```typescript
type MaybeAsync<T> = T | Promise<T>;

interface CacheType {
  init: () => MaybeAsync<void>,
  get: (key: string) => MaybeAsync<JSONSerializable>,
  set: (key: string, value: JSONSerializable) => MaybeAsync<void>,
  reset: () => MaybeAsync<void>
}
```

Reply:
🤔 Uhm I got this based on looking at [lib/multi-level.js:L18-24](https://github.com/godaddy/out-of-band-cache/blob/master/lib/multi-level.js#L19). I figured it was little too generic as well. But, what you are referring seems very similar to functions within `MemoryCache` [here](https://github.com/godaddy/out-of-band-cache/blob/master/lib/memory.js#L4). Is this what you were referring to?

```suggestion
/**
 * @typedef Cache
 * @prop {AsyncFunction} init Initialization function
 * @prop {AsyncFunction} get Item retrieval function
 * @prop {AsyncFunction} set Item storage function
 * @prop {AsyncFunction} reset Cache clear function
 */
```